ansible/roles/openshift_template_deployer/tasks/main.yml

```diff
@@ -0,0 +1,98 @@
+---
+# Ensure needed variables are set
+# This is a a place to verify needed values
+- name: Ensure that osvp is set
+  fail:
+    msg:  "Please ensure that 'osvp_<variables>' are set!"
+  when: item is undefined
+  with_items:
+  - "{{ ostd_template_name }}"
+  - "{{ ostd_template_file }}"
+  - "{{ ostd_project }}"
+  run_once: true
+
+# install rpm if needed aos_3.5_roles/openshift_hosted_templates/
+- name: install the "{{ ostd_rpm }}"
+  yum:
+    name: "{{ ostd_rpm }}"
+    state: installed
+  when: ostd_rpm
+
+- name: Make template changes if needed
+  yedit:
+    src: "{{ ostd_template_file }}"
+    key: "{{ item.key }}"
+    value: "{{ item.value }}"
+  when: ostd_template_edits
+  with_items: "{{ ostd_template_edits }}"
+
+  # TODO: Think about moving this role back into main playbook and remove it from here
+  # make master-config.yaml changes if needed to make sure master-config is up to date/truth
+- name: Configure the master config file with parameters needed for the provisioner pod
+  yedit:
+    src: /etc/origin/master/master-config.yaml
+    key: "{{ item.key }}"
+    value: "{{ item.value }}"
+    state: present
+    backup: False
+  with_items: "{{ ostd_master_config_edits }}"
+  when: ostd_master_config_edits
+
+  register: master_config_changes
+
+# restart master if needed
+# either here or in a handler
+- name: restart openshift master services
+  service:
+    name: "{{ item[1] }}"
+    state: restarted
+  delegate_to: "{{ item[0] }}"
+  with_nested:
+  - "{{ ansible_play_hosts }}"
+  - [ 'atomic-openshift-master-api', 'atomic-openshift-master-controllers' ]
+  run_once: true
+  when: master_config_changes.changed
+
+# pause if restarted
+- name: pause 10 seconds when services are restarted
+  pause:
+    seconds: 10
+  when: master_config_changes.changed
+
+# Enusre project is created
+- name: create project
+  oadm_project:
+    name: "{{ ostd_project }}"
+    node_selector: "{{ ostd_node_selector|default(omit)}}"
```

Comment:
Missing space before }} on several of these. @mwoodson suggested this as a style guideline.